bundle/config/mutator/resourcemutator/job_cluster_fixups.go

```diff
@@ -0,0 +1,75 @@
+package resourcemutator
+
+import (
+	"context"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+)
+
+type jobClustersFixups struct{}
+
+func JobClustersFixups() bundle.Mutator {
+	return &jobClustersFixups{}
+}
+
+func (m *jobClustersFixups) Name() string {
+	return "JobClustersFixups"
+}
+
+func (m *jobClustersFixups) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	for _, job := range b.Config.Resources.Jobs {
+		if job == nil {
+			continue
+		}
+		prepareJobSettingsForUpdate(&job.JobSettings)
+	}
+
+	return nil
+}
+
+// Copied from
+// https://github.com/databricks/terraform-provider-databricks/blob/a8c92bb/clusters/resource_cluster.go
+// https://github.com/databricks/terraform-provider-databricks/blob/a8c92bb130def431b3fadd9fd533c463e8d4813b/clusters/clusters_api.go#L440
+func ModifyRequestOnInstancePool(c *compute.ClusterSpec) {
+	// Instance profile id does not exist or not set
+	if c.InstancePoolId == "" {
+		// Worker must use an instance pool if driver uses an instance pool,
+		// therefore empty the computed value for driver instance pool.
+		c.DriverInstancePoolId = ""
+		return
+	}
+	if c.AwsAttributes != nil {
+		// Reset AwsAttributes
+		awsAttributes := compute.AwsAttributes{
+			InstanceProfileArn: c.AwsAttributes.InstanceProfileArn,
+		}
+		c.AwsAttributes = &awsAttributes
+	}
+	if c.AzureAttributes != nil {
+		c.AzureAttributes = &compute.AzureAttributes{}
+	}
+	if c.GcpAttributes != nil {
+		gcpAttributes := compute.GcpAttributes{
+			GoogleServiceAccount: c.GcpAttributes.GoogleServiceAccount,
+		}
+		c.GcpAttributes = &gcpAttributes
+	}
+	c.EnableElasticDisk = false
+	c.NodeTypeId = ""
+	c.DriverNodeTypeId = ""
+}
```

Comment:
This clears computed fields on the TF side. We don't have that problem here because we exclusively use the configuration as input to our requests. What doesn't work if we don't apply these fixups?

Reply:
Well, that would be a breaking change, so we probably should start with a warning. However, this is out of scope in this PR, here I'm just surfacing what TF already does so that we have parity between deployment backends.